### PR TITLE
Tooltip fix

### PR DESCRIPTION
**Pull request by dawinValenzuela**
**Checklist**

- [x] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://trello.com/c/pIdX7kME/43-borrower-flow-fixes-from-demo-june-19

**Storybook Link** (Link to a local storybook url to review new functionality)
https://storybook-tooltip-fix-7v6rokdl5q-uc.a.run.app/?path=/story/composer-primitive-tooltip--with-background-color

### CODE REVIEW

**Comment by tkanzakic** [comment]
we have a security issue in this PR but I have no access to review it, @stevejcox can you take a look please?

File: packages/components/package.json [hunk 1]
```diff
@@ -29,7 +29,9 @@
     "react-markdown": "^4.3.1",
     "react-modal": "^3.11.2",
     "react-tooltip": "^4.2.6",
-    "rimraf": "^3.0.2"
+    "rimraf": "^3.0.2",
+    "styled-components": "^5.1.1",
+    "twin.macro": "^1.4.1"
```
Comment (guilean) [inline packages/components/package.json:34]: Are we sure about that? Are we planning to use styled-components for styles from now on? @tkanzakic @stevejcox 

Adding three libraries to be able to change backgroundColor of tooltip seems overwhelming .

We might think a different approach to solve the original issue. Or we might commit to start working with css in js and use styled-components everywhere.

File: packages/components/src/components/primitives/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -16,30 +24,50 @@ type Props = {
...
+  className?: string;
 };
 
+const TooltipWrapper = styled(ReactTooltip)`
+  &:after {
+    ${({ background }) =>
+      background &&
+      (background === BackgroundColor.Primary100
+        ? tw`border-fill-primary-100`
+        : tw`border-fill-primary-75`)}
```
Comment (guilean) [inline packages/components/src/components/primitives/Tooltip/Tooltip.tsx:37]: Could we set `Primary100` or `Primary75` as default value for `backgroundColor` and skip this condition and just assign to background whatever `backgroundColor` prop is received?

File: packages/components/src/components/primitives/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -16,30 +24,50 @@ type Props = {
...
         id={id}
         place={place}
         effect={'solid'}
+        className={cn(backgroundColor, className)}
+        overridePosition={({ left, top }, _e, _t, node) => {
+          return {
+            top,
+            left: typeof node === 'string' ? left : Math.max(left, 0),
+          };
+        }}
```
Comment (guilean) [inline packages/components/src/components/primitives/Tooltip/Tooltip.tsx:67]: Looks like magic to me 🧙

File: packages/components/package.json (hunk 1 above)
Comment (stevejcox) [reply packages/components/package.json:34]: Chatted to @dawinValenzuela - going to remove styled components for now. I agree that if we want to go down that route, we should refactor to use that everywhere. And there are plenty of pros and cons for that approach, so we'd want to regroup about it first.

**Comment by stevejcox** [comment]
> we have a security issue in this PR but I have no access to review it, @stevejcox can you take a look please?

Will look into why you couldn't get access. The issue is `lodash`. One of its functions `zipObjectDeep` exposes a medium level security warning for `Prototype Pollution`. We don't use that function, so in theory is not an issue at all for us if we were going to continue to need these additions. The same issue exists for `lerna` which we use.

**Pull request closed without merging**